### ISSUE TITLE
Extensions update warns about duplicate packages

### ISSUE THREAD

**Issue by michaelaye** [jupyterlab/jupyterlab#6423]
**Describe the bug**

At the end of running `jupyter labextension update --all`
I get this:

```bash
WARNING in d3-array
  Multiple versions of d3-array found:
    1.2.4 ./~/d3-scale/~/d3-array from ./~/d3-scale/~/d3-array/src/index.js
    2.2.0 ./~/d3-array from ./~/d3-array/src/index.js


WARNING in jquery
  Multiple versions of jquery found:
    2.2.4 ./~/jupyter-matplotlib/~/jquery from ./~/jupyter-matplotlib/src/mpl_widget.js
    3.4.1 ./~/jquery from ./~/jquery-ui/ui/widget.js


WARNING in react
  Multiple versions of react found:
    16.4.2 ./~/react from ./~/react/index.js
    16.8.6 ./~/@jupyterlab/git/~/react from ./~/@jupyterlab/git/~/react/index.js


WARNING in react-dom
  Multiple versions of react-dom found:
    16.4.2 ./~/react-dom from ./~/react-dom/index.js
    16.8.6 ./~/@jupyterlab/git/~/react-dom from ./~/@jupyterlab/git/~/react-dom/index.js


WARNING in vega-lite
  Multiple versions of vega-lite found:
    2.5.1 ./~/vega-lite/build/src from ./~/vega-lite/build/src/compile/selection/selection.js
    2.6.0 ./~/vega-lite/build from ./~/vega-lite/build/src/index.js

Check how you can resolve duplicate packages:
https://github.com/darrenscerri/duplicate-package-checker-webpack-plugin#resolving-duplicate-packages-in-your-bundle

Child html-webpack-plugin for "index.html":
     1 asset
    Entrypoint undefined = index.html
    [KTNU] ./node_modules/html-loader!./templates/partial.html 567 bytes {0} [built]
    [YuTi] (webpack)/buildin/module.js 497 bytes {0} [built]
    [aS2v] ./node_modules/html-webpack-plugin/lib/loader.js!./templates/template.html 1.22 KiB {0} [built]
    [yLpj] (webpack)/buildin/global.js 489 bytes {0} [built]
        + 1 hidden module
Child worker:
                                 Asset      Size  Chunks             Chunk Names
        21c506c264866aa2afec.worker.js  1.45 MiB    main  [emitted]  main
    21c506c264866aa2afec.worker.js.map  1.73 MiB    main  [emitted]  main
    Entrypoint main = 21c506c264866aa2afec.worker.js 21c506c264866aa2afec.worker.js.map
    [yLpj] (webpack)/buildin/global.js 489 bytes {main} [built]
        + 2 hidden modules
   Done in 22.74s.
```
**To Reproduce**

1. Install some jupyterlab extensions
2. run `jupyter labextension update --all`

**Expected behavior**

No duplicate package warnings.

**Desktop (please complete the following information):**

- OS: macOS 10.14.3
- Browser Safari 12.0.3
- JupyterLab 0.35.6

**Comment by vidartf** [comment]
Thanks for the report!

You state the expected behavior as "No duplicate package warnings". Would you mind expanding a little on this? E.g.:
- Would you expect lab to simply not print the warnings?
- Would expect lab to resolve the duplication somehow?
- Would expect the duplication to lead to an error?

**Comment by michaelaye** [comment]
Good question. My confusion and worry resulted from the fact that there was a solution indicated how to resolve duplicate packages, which raised the question if it was on me to perform this, or, alternatively, if this is something that would solve itself in a future update.
Also I have seen before, b/c I like to live on the edge with extensions and stuff, that a paradigm or framework changed, so that the respective code was now installed in more than one place (kernels before conda_nb_kernels come to mind), and the only way to clean up that mess would have been to reinstall either all the framework or even the conda env (which is fortunately quite fast with a few bash scripts).
To directly answer your questions:

> * Would you expect lab to simply not print the warnings?

If the author of the extension or lab is aware of a temporary warning due to multiple installs that will be cleaned up in a future update/upgrade, then yes, I'd expect not to see them as end-user.

> * Would you expect lab to resolve the duplication somehow?

That would be ideal, and is kind of expected by me, b/c I never personally touched any of that code, I always simply followed the `jupyter labextension install xxx` instruction, and wouldn't even know where stuff is being installed.

> * Would you expect the duplication to lead to an error?

As explained above, the advise on how the duplication could be resolved kinda made me unsure if I need to act now. I'm guessing this advise is expressed from the underlying framework and not by extensions authors or lab itself, but I'm a fairly technical science user and if I'm confused, less technically inclined will be as well, I reckon.

**Comment by jasongrout** [comment]
Great discussion. It seems that the duplicate packages here are coming from different third-party extensions requiring slightly different versions of common dependencies. It looks like the error is coming from the build system (not us, but the build tool we use).

Generally these errors should be fine to ignore.

Resolving these duplicate packages would be a difficult issue, as it requires the third-party extension authors to coordinate their dependency versions.

**Comment by vidartf** [comment]
> It seems that the duplicate packages here are coming from different third-party extensions requiring slightly different versions of common dependencies.

Well, many of the duplicates are major version different, which might be an issue if objects cross extension borders. It might also be an issue with e.g. jQuery, which generally recommend to *not* have two versions on the same page.

For now, I think the guidance "probably safe to ignore, but might be the cause of undefined errors" is probably the most correct. As such, I do not think we can safely silence these warnings, but rather we should consider if we can present the warnings in a more constructive way for end users.

**Comment by michaelaye** [comment]
Upon further inspection of above errors I pasted I actually don't understand the printed out paths?
In none of my machines do I have a `~/react-dom` folder? And what path is @jupyterlab ??

```
 Multiple versions of react-dom found:
    16.4.2 ./~/react-dom from ./~/react-dom/index.js
    16.8.6 ./~/@jupyterlab/git/~/react-dom from ./~/@jupyterlab/git/~/react-dom/index.js
```

**Comment by vidartf** [comment]
Since #6567, these warnings (that are expected) will not be displayed by default. Closing as resolved in 1.0.
